plrust/src/pgproc.rs

```diff
@@ -76,6 +76,28 @@ impl PgProc {
         }
     }
 
+    #[inline]
+    fn cmin(&self) -> pg_sys::CommandId {
+        // SAFETY:  self.inner will be valid b/c that's part of what pg_sys::SearchSysCache1()
+        // does for us.  Same is true for t_data
+        unsafe {
+            self.inner
+                .as_ref()
+                .t_data
+                .as_ref()
+                .unwrap_unchecked() // SAFETY: t_data will never be null and `cmin()` is called in a potentially hot path
+                .t_choice
+                .t_heap
+                .t_field3
+                .t_cid
+        }
+    }
+
+    #[inline]
+    pub(crate) fn generation_number(&self) -> u64 {
```

Comment:
Couldn't decide if we wanted a stronger type than `u64`.  I considered `struct GenerationNumber { xmin: TransactionId, cmin: CommandId }` but didn't want to do all that typing if this idea turns out bad.

Reply:
A u64 is fine for an incrementing counter, we can lock things down later if needs be.